src/cursor/cur_stat.c

```diff
@@ -383,6 +383,75 @@ __curstat_conn_init(WT_SESSION_IMPL *session, WT_CURSOR_STAT *cst)
     cst->stats_desc = __wt_stat_connection_desc;
 }
 
+/*
+ * __init_layered_constituent_stats --
+ *     For either an ingest or stable table, aggregate dhandle stats into a stat cursor.
+ */
+static int
+__init_layered_constituent_stats(WT_SESSION_IMPL *session, WT_CURSOR_STAT *cst, bool *stats_cleared)
+{
+    WT_DATA_HANDLE *dhandle;
+    WT_DECL_RET;
+
+    dhandle = session->dhandle;
+
+    if ((ret = __wt_btree_stat_init(session, cst)) == 0) {
+        /*
+         * Only clear the aggregate stats if (1) we initialized this tree's stats, and (2) another
+         * constituent call didn't do it.
+         */
+        if (!*stats_cleared)
+            __wt_stat_dsrc_init_single(&cst->u.dsrc_stats);
+        *stats_cleared = true;
+
+        __wt_stat_dsrc_aggregate(dhandle->stats, &cst->u.dsrc_stats);
+        if (F_ISSET(cst, WT_STAT_CLEAR))
+            __wt_stat_dsrc_clear_all(dhandle->stats);
+    }
+
+    return (ret);
+}
+
+/*
+ * __curstat_layered_init --
+ *     Initialize the statistics for a layered table.
+ */
+static int
+__curstat_layered_init(
+  WT_SESSION_IMPL *session, const char *uri, const char *cfg[], WT_CURSOR_STAT *cst)
+{
+    WT_DATA_HANDLE *dhandle;
+    WT_DECL_RET;
+    WT_LAYERED_TABLE *layered;
+    bool stats_cleared;
+
+    stats_cleared = false;
+
+    WT_RET(__wt_session_get_btree_ckpt(session, uri, cfg, 0, NULL, NULL));
+    dhandle = session->dhandle;
+    WT_ASSERT(session, dhandle->type == WT_DHANDLE_TYPE_LAYERED);
+    layered = (WT_LAYERED_TABLE *)dhandle;
+
+    /* Do the ingest table. */
+    dhandle = layered->ingest;
+    WT_WITH_DHANDLE(
+      session, dhandle, ret = __init_layered_constituent_stats(session, cst, &stats_cleared));
+    WT_RET(ret);
+
+    /* Now do the stable table. */
+    dhandle = layered->stable;
+    WT_WITH_DHANDLE(
+      session, dhandle, ret = __init_layered_constituent_stats(session, cst, &stats_cleared));
+    WT_RET(ret);
+
```

Comment:
This is fine for now.  Perhaps eventually we'll want knowledge of ingest and stable to be kept within the cur_layered.c .  I'm thinking if we someday have multiple ingest tables, we'll need to do this differently.

Reply:
Yeah I think there's an interesting design discussion there. Although I'm not sure that having multiple ingest tables changes anything, since presumably the ingest part of the stats would just be for the single most recent table?